File: TBZRMTO/BLL/Common/PermissionGroup_T/BLLPermissionGroup_T.cs
using System;
using System.Collections.Generic;
using System.Text;
using HPS.BLL.BusinessObjectMain;

namespace HPS.BLL.PermissionGroupBLL{

public class BLLPermissionGroup_T
: BusinessObjectBase{

public enum PermissionGroup_TField {
PermissionGroupID_int
,PermissionGroup_nvc
,ParentID_int
} 


private Int32 _PermissionGroupID_int;
       public Int32 PermissionGroupID_int
        {
            get
            {
                return _PermissionGroupID_int;
            }
            set
            {
                this._PermissionGroupID_int = value;
            }
        }


private String _PermissionGroup_nvc;
       public String PermissionGroup_nvc
        {
            get
            {
                return _PermissionGroup_nvc;
            }
            set
            {
                this._PermissionGroup_nvc = value;
            }
        }


private Nullable< Int32> _ParentID_int;
       public Nullable< Int32> ParentID_int
        {
            get
            {
                return _ParentID_int;
            }
            set
            {
                this._ParentID_int = value;
            }
        }



public override void AddValidationRules(){
ValidationRules.AddRules(new Hepsa.Core.Validation.NotNullRule("PermissionGroup_nvc","عنوان گروه دسترسی"));
ValidationRules.AddRules(new Hepsa.Core.Validation.StringMaxLengthRule("PermissionGroup_nvc","عنوان گروه دسترسی",100));
ValidationRules.AddRules(new Hepsa.Core.Validation.RelationRule("PermissionGroupID_int","","PermissionGroup_T","PermissionGroupID_int"));
ValidationRules.AddRules(new Hepsa.Core.Validation.UniqueRule("UniquePermissionGroupID_int","شناسه گروه دسترسی","PermissionGroup_T","PermissionGroupID_int"));
}
}
}

File: HepsaCore/BLL/Common/PermissionGroup_T/BLLPermissionGroup_T.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

Namespace BLL

    Public Class BLLPermissionGroup_T
        Inherits BLL.BusinessObjectBase

        Public Enum PermissionGroup_TField
            PermissionGroupID_bint
            PermissionGroup_nvc
            SystemID_tint
        End Enum


        Private _PermissionGroupID_bint As Int64
        Private _PermissionGroup_nvc As String
        Private _SystemID_tint As Nullable(Of Byte)


        Public Property PermissionGroupID_bint() As Int64
            Get
                Return _PermissionGroupID_bint
            End Get
            Set(ByVal value As Int64)
                Me._PermissionGroupID_bint = value
            End Set
        End Property

        Public Property PermissionGroup_nvc() As String
            Get
                Return _PermissionGroup_nvc
            End Get
            Set(ByVal value As String)
                Me._PermissionGroup_nvc = value
            End Set
        End Property

        Public Property SystemID_tint() As Nullable(Of Byte)
            Get
                Return _SystemID_tint
            End Get
            Set(ByVal value As Nullable(Of Byte))
                Me._SystemID_tint = value
            End Set
        End Property

        Friend Overloads Overrides Sub AddValidationRules()
            ValidationRules.AddRules(New Validation.NotNullRule("PermissionGroupID_bint", "کد گروه مجوز"))
            ValidationRules.AddRules(New Validation.NumericRule(Of Int64)("PermissionGroupID_bint", "کد گروه مجوز", 1, Nothing, 4, 0))
            ValidationRules.AddRules(New Validation.NotNullRule("PermissionGroup_nvc", "عنوان گروه مجوز"))
            ValidationRules.AddRules(New Validation.StringMaxLengthRule("PermissionGroup_nvc", "عنوان گروه مجوز", 100))
            ValidationRules.AddRules(New Validation.RelationRule("RelationSystemID_tint", "سیستم", "System_T", "SystemID_tint"))
            ValidationRules.AddRules(New Validation.UniqueRule("UniquePermissionGroupID_bint", "کد گروه مجوز", "PermissionGroup_T", "GroupID_bint"))
            ValidationRules.AddRules(New Validation.UniqueRule("UniquePermissionGroup_nvc", "عنوان گروه مجوز", "PermissionGroup_T", "PermissionGroup_nvc"))
        End Sub

    End Class
End Namespace